autopush/endpoint.py

```diff
@@ -254,6 +273,49 @@ def _db_error_handling(self, d):
         d.addErrback(self._response_err)
         return d
 
+    def _extract_jwt(self, token, crypto_key):
+        # first split and convert the jwt.
+        sig = None
+        payload = {}
+
+        def fix_padding(string):
+            return string + '===='[len(sig) % 4:]
+
+        try:
+            (ehead, epayload, sig) = token.split('.', 2)
+            sig = fix_padding(sig)
+        except ValueError:
+            raise VapidAuthException("Invalid bearer token")
+        head = json.loads(base64.urlsafe_b64decode(ehead))
+        payload = json.loads(base64.urlsafe_b64decode(epayload))
+        if sig and crypto_key:
+            key = base64.urlsafe_b64decode(fix_padding(crypto_key))
+            vk = ecdsa.VerifyingKey.from_string(key, curve=ecdsa.NIST256p)
+            try:
+                jws.verify(head, payload, sig, vk)
+            except Exception:
+                # failure. Probably should log this...
```

Comment:
Since this is a blanket exception, it should be logged. Specific exceptions should be caught when possible.